docs/changelog/4_x.rst

```diff
@@ -0,0 +1,467 @@
+  {b'\x05\x82\xd8YLF\xe7\xd4\x12\x91\n\xdb$\xf1!v\xf9\xd4\x89\xc4'}
+  edgedb> select ext::pgcrypto::gen_salt('md5');
+  {'$1$FjNlXgX7'}
+
+Standard algorithms are "md5", "sha1", "sha224", "sha256", "sha384" and
+"sha512". Moreover, any digest algorithm OpenSSL supports is automatically
+picked up.
+
+
+pg_trgm
+-------
+
+This extension provides a bunch of functionality used to determine string
+similarity, which makes it a good text search alternative for some use cases:
+
+edgedb> with x := {'hello world', 'word hero', 'help the world'}
+....... select res := (x, ext::pg_trgm::word_similarity(x, 'hello world'))
+....... order by res.1 desc;
+{('hello world', 1), ('help the world', 0.5), ('word hero', 0.35714287)}
+
+
+
+Additional changes
+==================
+
+EdgeQL
+------
+
+* Add ``anyobject`` pseudo type.
+  (:eql:gh:`#5961`)
+
+  It is a placeholder mainly used in polymorphic functions. It denotes any
+  object type and like :eql:type:`anytype` it preserves the actual type if the
+  placeholder ``anyobject`` appears in the function return signature.
+
+* Add new style of ``if``/``then``/``else`` syntax.
+  (:eql:gh:`#6074`)
+
+  Many people find it more natural to write "if ... then .. else ..." for
+  conditional expressions because it mirrors the conditional statement from
+  other familiar programming languages.
+
+  .. code-block:: edgeql-repl
+
+    edgedb> select if count(Object) > 0 then 'got data' else 'no data';
+    {'got data'}
+
+* Support conditional DML.
+  (:eql:gh:`#4437`)
+
+  It can be useful to be able to create, update or delete different objects
+  based on some condition:
+
+  .. code-block:: edgeql
+
+    with
+      name := <str>$0,
+      admin := <bool>$1
+    select if admin then (
+        insert AdminUser { name := name }
+    ) else (
+        insert User { name := name }
+    )
+
+  A different use-case of conditional DML is using a :eql:op:`coalesce`
+  operator to express things like "select or insert if missing":
+
+  .. code-block:: edgeql
+
+    select (select User filter .name = 'Alice') ??
+           (insert User { name := 'Alice' });
+
+* Add ``contains`` for JSON so that it can be used with ``pg::gin`` index.
+  (:eql:gh:`#5703`)
+
+* Add :eql:func:`to_bytes` to convert :eql:type:`str` into :eql:type:`bytes`
+  using UTF-8 encoding.
+  (:eql:gh:`#4281`)
+
+* Add :eql:func:`to_str` to convert :eql:type:`bytes` into :eql:type:`str`
+  using UTF-8 encoding.
+  (:eql:gh:`#4281`)
+
+* Add ``enc::base64_encode`` and ``enc::base64_decode`` functions.
+  (:eql:gh:`#5963`)
+
+  .. code-block:: edgeql-repl
+
+    edgedb> select enc::base64_encode(b'hello');
+    {'aGVsbG8='}
+    edgedb> select enc::base64_decode('aGVsbG8=');
+    {b'hello'}
+
+* Support optional config values.
+  (:eql:gh:`#6025`)
+
+* Add ``when`` clause to triggers to enable them to be conditional.
+  (:eql:gh:`#5994`)
+
+* Allow empty arrays without cast in ``insert``.
+  (:eql:gh:`#6218`)
+
+
+
+Bug fixes
```

Comment:
We don't normally list bugfixes that have been backported to an earlier release, which, I assume, all have been. @msullivan ?

Reply:
I think I cross-checked and at least the fixes here are not mentioned in the 3_x changelog (I pruned a bunch of fixes that were backported and mentioned already).

Reply:
A bunch of these were not backported, but a bunch of them were. (All of the GraphQL ones, for example.)

Also, we usually use the PR numbers for these, not the issue numbers.

There is a that I use to help generate these that Elvis gave me that we should put somewhere real